[Brian Fung] Duke Increments

briyanii opened the pull request:
My week 3 increments for Duke individual project.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

briyanii committed:
Greet, Echo, Exit

briyanii committed:
Add, List

briyanii committed:
Mark as Done, and Classes

briyanii committed:
ToDo, Event,Deadline

briyanii committed:
ToDos, Events, Deadlines

briyanii committed:
ToDos, Events, Deadlines

briyanii committed:
Level-5

Added exceptions.
Added error handling.
Added input validation (not very extensive).

briyanii committed:
Level-6

Added support for deleting tasks from the list.

briyanii committed:
Level-6

Added support for deleting tasks from the list.

briyanii committed:
A-TextUiTesting

Added input/output redirection bash script to semi-automate testing of Duke
Modified bash script for use with different input/output pairs

briyanii committed:
deleted text-ui-test/input.txt

deleted redundant test input file

briyanii committed:
branch-Level-7

Added support for saving the task list between sessions
Added a sample list (src/main/java/DukeSaveFile.txt)

briyanii committed:
branch-Level-8

Added support for parsing time in "dd/MM/yyyy HHmm" pattern (strictly in this pattern without deviation).

briyanii committed:
SaveLoad

Added functionality to TaskList for loading and saving to/from different files (actual path still hard coded into Duke.java)
Added basic exceptions for Loading Saving from file path/file errors

briyanii committed:
Save Load fixed

changes made to Duke.java and TaskList.java were not properly staged in previous commit

briyanii committed:
Date and Time

Fixed incorrect integer ordinal postfix when parsing day of month

briyanii committed:
Merge branch 'branch-Level-7'

Merged load/save functionality into master

briyanii committed:
Merge branch 'branch-Level-8'

Merged date and time parsing (of Deadline and Event task parameter) functionality into master

briyanii committed:
A-MoreOOP

extracted closely related code into same classes

	new file:   UserInterface.java
	- deals with iteraction between user and duke

 	modified:   Parser.java
	- deals with interpreting/filtering user input into information useable by duke

 	modified:   Output.java
	- deals with printing output

	new file:   Storage.java
	- extracted loading saving functionality to upper level class
	- handles file reading and writing

	modified:   Command.java
	- changed Command from enum to class
	- changed to containers for commands and parameters

	new file:   AddTaskCommand.java
	- extracted commands for making different task to an upper level class

	new file:   CompleteTaskCommand.java
	- extracted done command to an upper level class

	new file:   DeleteTaskCommand.java
	- extracted delete command to an upper level class

	new file:   ShowListCommand.java
	- extracted list command to an upper level class

	new file:   ExitCommand.java
	- extracated bye command to an upper level class

 	new file:   SaveFiles/DukeSave01.txt
	- save file for content of list between sessions

briyanii committed:
A-Packages

grouped related classes together:
-classes meant for I/O
-classes related to Command
-classes related to Tasks

briyanii committed:
Merge branch 'gradle'

Added gradle wrapper to project

briyanii committed:
A-JUnit

Added support for Unit Tests with JUnit
- tests for Parser methods;
- tests for some UserInterface methods;

Fixed incorrect output from Parser.parseDateTime
- the input "12/12/1212 1212" now will now properly return "12th of DECEMBER 1212, 12:12pm" instead of "12th of DECEMBER 1212, 0:12am"

briyanii committed:
ShadowJar

Added ShadowJar plugin to project to help create .jar executable
'com.github.johnrengelman.shadow' version '5.1.0'

briyanii committed:
Level-9: Find

Added support for searching for a keyword  in tasklist

briyanii committed:
Level-9

briyanii committed:
A-codingStandard

cleaned up code to match coding standard (less comments)

briyanii committed:
branch-A-CodingStandard

briyanii committed:
branch-A-JavaDoc

Added JavaDoc comments to the source files
Deleted some unused classes

briyanii committed:
Merge branch 'branch-A-JavaDoc'

Merged branch with JavaDoc comments into master branch

briyanii committed:
Merge branch 'branch-A-CodingStandard'

briyanii committed:
A-CodingStandard

Merged modification to code to conform to coding standards into master branch
Fixed some parts where it did not conform to coding standards in the master branch

briyanii committed:
Added JavaDocs to some methods that were added seperately of the A-JavaDoc branch

briyanii committed:
Merge branch 'branch-Level-9'

Merged find functionality into master branch from branch-Level-9
Resolved merge conflicts when from merging branches

briyanii committed:
Fixed bug: missing 'break;' statement

Added missing break statement after switch case which caused find to not be recognised as a valid command

briyanii committed:
Removed text-ui-testing:

removed input and expected output files

briyanii committed:
Remove final attribute from two fields in Command class

briyanii committed:
Add prototype GUI using JavaFX

Add basic GUI
Refactor instances Type enum to have "TYPE_" prefix
Remove Input class (no longer needs console input)
Remove Output class (output no longer needs to be formatted for console output)

bernicechio commented:
- I appreciate how your parser test case is very detailed
- Code is well written 👍🏻
- Good use of lamda expressions and enumeration

CowSaysBaa commented:
I am no expert at software engineering but here's my 2 cents.
Code is well formatted. The parsing operation is quite unintuitive but unique. To be honest, the code is rather complicated, which makes reading and understanding very challenging, but I can appreciate the effort that went into the abstraction of data and the preparation for changes in the future, especially for the Type enum which I guess guarantees type safety.

briyanii committed:
Use FXML for JavaFX scene layouts

Use .fxml files for the basic design of the GUI (from the provided tutorial)

Repackage classes to be more logical
- e.g. Parser in duke.command instead of duke.io

Remove redundant classes
- Input, Output: no longer using command line for input and output

Add JUnit tests for new methods for Duke.

briyanii committed:
Merge branch 'Level-10'

Add GUI support to Duke

briyanii committed:
Add checkstyle plugin to Gradle.

Use the provided checkstyle.xml file from gradle branch to check for coding style violations using checkstyle plugin.

briyanii committed:
Make GUI window non-resizeable

CowSaysBaa reviewed (commented):
I am no expert at software engineering but here's my 2 cents.
Code is well formatted. The parsing operation is quite unintuitive but unique. To be honest, the code is rather complicated, which makes reading and understanding very challenging, but I can appreciate the effort that went into the abstraction of data and the preparation for changes in the future, especially for the Type enum which I guess guarantees type safety.

briyanii committed:
Fix GUI to properly display file missing error

GUI now displays whether the TaskList can be loaded from the specified file path.
If the file path is invalid, and cannot be made, Duke now deactivates.

briyanii committed:
Add assertions to branch-A-Assertions

Add assertions to check that arguments that should not be null should be non-null.
Add assertions that delete, complete commands should have int argument > 0.

briyanii committed:
Merge pull request #6 from briyanii/branch-A-Assertions

Merge branch-A-Assertions into master

briyanii committed:
Fix JUnit test regressions caused by assertions

Fix JUnit tests methods which were broken by introducing Java assertions.

Remove some redundant assertions

briyanii committed:
Merge pull request #7 from briyanii/branch-A-Assertions

Fix JUnit test regressions caused by assertions

briyanii committed:
Add Command for Relaxed Search

Add RelaxedSearchCommand to allow for searching for tasks words similar to keyword

briyanii committed:
Refactor code to improve code quality (#8)

* Constrict access modifiers

* Remove Assignments to Parameters

* Extract method to write lines to file

Extract method to write lines to file from the 3 writeFromTask methods in Storage.

* Remove redundant String initialization

* Remove dead code

Remove unused import statement in duke.gui.MainWindow
Remove unused class DukeEmptyListException in duke.tasklist

* Extract method to display good bye popup.

Extract method to display good bye popup from start method in Gui.java

* Use unqualified class names.

* Replace if-else with ternary

* Replace while-loop with for-loop

* Replace String literals with symbolic constant

Replace String literals with symbolic constants for flags used indicate
task type and completion status in save file.

* Extract methods from Parser.parseAsCommand

Extract method extractCommandType to get Command Type from user input

Extract method extractArguements to get arguments provided in user input

Add guard clause to return 0 parameter Commands after extracting Command Type

* Extract method to display multiple DialogBox

Extract method "display" to display multiple DialogBox

* Extract constants in DialogBox.java

Extract constants for insets, corner radii, and color arguments for DialogBox

* Change method declaration

Change static method to get number of arguments expected to instance method, and
Rename method

Change static method to get delimiters separating arguments to instance method, and
Rename method

Change static method to get parameter names to instance method, and
Rename method

* Change static method to instance method

Change static method getArgumentsUsed to instance method

Change static method getTypeOf to instance method
Rename method getTypeOf to getType

Change switch String literal case to enum

* Rename Variables

Rename 1 and 2 char-length variables with more readable names

briyanii committed:
Fix relaxfind not properly displaying results

briyanii committed:
Fix save/load feature.

Previously task list save file would be read from and written to resource directory,
this causes Duke to be unable to save task list when run from jar as the jar is locked.

Change so that a new directory in the directory the jar is contained in to be created,
and save files read to and written from there.

briyanii committed:
Merge branch 'branch-fix-IO'

briyanii committed:
Add Load and Save commands and modify README.md

briyanii committed:
Merge branch 'branch-load-save-2'

briyanii committed:
Change version in build.gradle to match actual version

briyanii committed:
Fix links in README.md

briyanii committed:
Fix READ.md links

Actually fix the links this time

briyanii committed:
Set theme jekyll-theme-cayman

briyanii committed:
Fix README.md

briyanii committed:
Delete _config.yml

briyanii committed:
Merge branch 'master' of https://github.com/briyanii/duke

briyanii committed:
Update README.md

briyanii committed:
Fix checkstyle violations

briyanii committed:
Fix getResource not properly loading resource files

and increase dimensions of duke

briyanii committed:
Fix Duke not properly creating files when the directory has spaces in its name.

briyanii committed:
Add links to feature subsections at the start of README.md

briyanii committed:
Remove anchor links in README.md

Not working
